src/routes/versions.rs

```diff
@@ -167,3 +358,116 @@ pub async fn version_delete(
         Ok(HttpResponse::NotFound().body(""))
     }
 }
+
+// under /api/v1/file/{hash}
+#[get("{version_id}")]
+pub async fn get_version_from_hash(
+    info: web::Path<(String,)>,
+    pool: web::Data<PgPool>,
+) -> Result<HttpResponse, ApiError> {
+    let hash = info.into_inner().0;
+
+    let result = sqlx::query!(
+        "
+        SELECT version_id FROM files
+        INNER JOIN hashes ON hash = $1
```

Comment:
There are no hash algorithms with the same length, so this works fine

Reply:
It may work, but it's still wrong.  This must either only use one hash algorithm or have the route specify the hash in some way (ex. prefixing with `md5` or `sha1`).